packages/core/aqua-to-js/src/generate/converters.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @typedef {import("@fluencelabs/js-client").NonArrowSimpleType} NonArrowSimpleType
+ * @typedef {import("@fluencelabs/js-client").JSONValue} JSONValue
+ */
+
+/**
+ * Convert value from its representation in aqua language to representation in typescript
+ * @param {JSONValue} value - value as represented in aqua
+ * @param {NonArrowSimpleType} schema - definition of the aqua schema
+ * @returns {JSONValue} value represented in typescript
+ */
+export function aqua2ts(value, schema) {
+  if (schema.tag === "nil") {
+    return null;
+  } else if (schema.tag === "option") {
+    if (!Array.isArray(value)) {
+      throw new Error("Bad schema");
```

Comment:
I think a better error message could be useful, if you think a bit about when this error pottentially can happen. I would create an error text that explains what went wrong in general (e.g. `Wasn't able to convert aqua to js`) and then for each case some particulars that could be useful for debugging (e.g.`schema generated by the aqua comiler expects an option and options are represented as js arrays but found type: ${typeof value}:  ${value}`) If `aqua2ts` recursive function also accepts the third argument which is `context` - then you can also store some context that can be useful to display in the error so it says how deep the error accured e.g. `error happened at: someProperty.someNestedProperty.someNetedArray[0]`

Reply:
These errors are impossible actually. Only if you write your own wrong schema by hand and use it. But this is not the case. I think that no need to spend much efforts improving this. This made for satisfying `ts`, not for giving decent error messages.

Reply:
These errors are tottally possible if something goes wrong with the compiler and for guys who write the compiler at least a little effort here would be useful to understand where the problem happened

Reply:
Well, probably we can do some runtime validation here